main/test/src/mill/main/MainModuleTests.scala

```diff
@@ -27,6 +29,16 @@ object MainModuleTests extends TestSuite {
       Map("1" -> "hello", "2" -> "world")
     }
     def helloCommand(x: Int, y: Task[String]) = Task.Command { (x, y(), hello()) }
+
+    /**
+     * The hello worker
+     */
+    def helloWorker = Task.Worker {
+      // non-serializable, but everything should work fine nonetheless
+      new AutoCloseable {
+        def close() = ()
```

Comment:
We should make sure `.close()` ends up being called, because otherwise you end up leaking workers/classloaders/filehandles which can be pretty subtle and hard to debug. Some hacky global registry of "things we closed" would suffice for test suite purposes

Reply:
The `clean-worker` tests in that file do just that